### DISCUSSION TITLE
Device Export Template- Parent Bay

### DISCUSSION THREAD

**Discussion by mkarel** [netbox-community/netbox#13235] [Q&A] [answered]
I am creating a device export template. I am trying to include the devices parent device name using the following code {{ device.parent_bay.device.name }}. When I runt the export I get the error "_There was an error rendering the selected export template (Device Export): 'dcim.models.devices.Device object' has no attribute 'parent_bay_'" 

When I look at the nbshell and run that query, it returns the parent device name. What am I doing wrong.  Thank in advance.

Template Code:
{% for device in queryset %}
Name: {{ device.name }}
Status: {{ device.status }}
Site: {{ device.site}}
Location: {{ device.location}}
Rack: {{ device.rack}}
Device Height: {{ device.device_type.u_height }}
Role: {{ device.device_role }}
Manufacturer: {{ device.device_type.manufacturer}}
Type: {{ device.device_type }}
Serial number: {{ device.serial }}
**Parent Device: {{ device.parent_bay.device.name }}**
Parent Device Bay: {{ device.parent_bay }}
Tags: {{ device.tags.names() | join(", ") }}
{% endfor %}

**Comment by zmaster7** [comment, marked as answer]
Are all the devices you are exporting "child" devices? My guess is you have some devices matching your filter that aren't "child" devices, which explains why the field doesn't exist on those. To prevent it from erroring, I believe you just need an if statement to check for device.parent_bay.

Example:
```
{%- if device.parent_bay -%}
Parent Device: {{ device.parent_bay.device.name }}
Parent Device Bay: {{ device.parent_bay }}
{%- endif -%}
```


If that doesn't work, you can check to see if it is a child device:

```
{%- if device.device_type.subdevice_role == "child" -%}
Parent Device: {{ device.parent_bay.device.name }}
Parent Device Bay: {{ device.parent_bay }}
{%- endif -%}
```

**Reply by mkarel** [reply]
@zmaster7 Thank you so much. It did not cross my mind that a non-parent/child device would cause an issue. here is my working template code.

{% for device in queryset %}
Name: {{ device.name }}
Status: {{ device.status }}
Site: {{ device.site}}
Location: {{ device.location}}
Rack: {{ device.rack}}
Device Height: {{ device.device_type.u_height }}
Role: {{ device.device_role }}
Manufacturer: {{ device.device_type.manufacturer}}
Type: {{ device.device_type }}
Serial number: {{ device.serial }}
{% if device.parent_bay -%}
Parent Device: {{ device.parent_bay.device.name }}
Parent Device Bay: {{ device.parent_bay }}
{%- else -%}
Parent Device:
Parent Device Bay:
{%- endif %}
Tags: {{ device.tags.names() | join(", ") }}
{% endfor %}
